StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/Vertical Main Screen/VerticalMandateView.swift

```diff
@@ -34,18 +38,20 @@ final class VerticalMandateView: UIView {
         // Remove old mandate view, if any
         mandateView?.removeFromSuperview()
         mandateView = nil
-        guard let paymentMethodType else {
+        guard let state, let paymentMethodType = state.paymentMethodType else {
             return
         }
 
         // Generate the form
-        guard
-            let form = formProvider(paymentMethodType),
-            !form.collectsUserInput
-        else {
-            // If it collects user input, the mandate will be displayed in the form and not here
+        guard let form = formProvider(paymentMethodType) else {
             return
         }
+
+        if form.collectsUserInput && !state.isReusing {
+            // If it collects user input and is not being re-used, the mandate will be displayed in the form and not here
+            return
+        }
```

Comment:
What does isReusing mean? Does it mean like, whether or not we have a saved payment method selected? Maybe rename `isSavedPaymentMethod`?

IMO we should branch based on if it's a saved PM or not. If it's not, trigger the old form-based code. If it is, switch on the payment method type.  The "generate form" logic just doesn't hold up for saved PMs. 

```
if isSavedPaymentMethod {
  switch paymentMethodType {
   ... manually build mandates
  }
} else {
  ... original form-generating code
}
```